### PR TITLE
Simplifying

### PR DESCRIPTION
**Pull request by fearphage**
Using `String#lastIndexOf` instead of `String#charCodeAt` to simplify the code and remove looping.

[Review on Reviewable](https://reviewable.io/reviews/mathiasbynens/string.prototype.endswith/7)

### CODE REVIEW

File: endswith.js [hunk 1]
```diff
@@ -36,16 +33,8 @@ if (!String.prototype.endsWith) {
 			}
```
Comment (mathiasbynens) [inline endswith.js]: If `this == null` then a `TypeError` must be thrown. Your patch would fail if `this` is `{ toString: function() { throw new RangeError(); } }`.

Comment (mathiasbynens) [inline endswith.js]: Could you add a test for this?

Comment (fearphage) [inline endswith.js]: There is already a [passing test](/fearphage/String.prototype.endsWith/blob/12b27ffd91f6660f525f4b8d63b22808d0d7b5d5/tests/tests.js#L236) for the `RangeError`.

Comment (mathiasbynens) [inline endswith.js]: Do you understand why? I don’t. Looks like `assertThrows` is broken?

Comment (fearphage) [inline endswith.js]: Yes, the call to `String(this)` triggers the `RangeError`. That's the first thing that takes place. Then it checks to see if `this` is `null` or `search` is a regular expression.

Comment (mathiasbynens) [inline endswith.js]: Ah, I was confused when I made my earlier comments. Will remove them

I should have said:

> With this patch, if `this == null` but `String(this)` throws e.g. a `RangeError`, then the `TypeError` (because `this == null`) isn’t thrown.

But there can be no value `this` that is `== null` for which `String(this)` throws.

Comment (fearphage) [inline endswith.js]: You are absolutely correct. I couldn't write a failing test for this however. Is this a corner case worth worrying about?

Comment (mathiasbynens) [inline endswith.js]: Probably not. But the current code is more direct translation of the steps in the specced algorithm and might therefore be considered to be more readable (it is IMHO).

Comment (fearphage) [inline endswith.js]: And by current code, you're referring to your code prior to my change, correct?

I think the current state is a bit more readable and why I made the change. One thing that confused me about the old code is your check for the truthiness of the `search` value.

```` javascript
if (search && toString.call(search) == '[object RegExp]') {
````

Is that a part of spec? Can you direct me to it?

Comment (fearphage) [inline endswith.js]: I generally don't agree with writing code for the purpose of writing code. With a failing test, I would revert this immediately. As it stands, the case you were previously accounting for was untested.

**Comment by ljharb** [comment]
I think it's better to keep this matching the spec as precisely as possible.
